tests/unit/test_loader_legacy.py

```diff
@@ -0,0 +1,311 @@
+
+
+def test_load_metadata_missing_yaml(populated_role_root):
+    os.unlink(os.path.join(populated_role_root, "meta", "main.yml"))
+
+    with pytest.raises(exc.ImporterError, match="Metadata not found"):
+        LegacyRoleLoader(populated_role_root, "my-namespace")._load_metadata()
+
+
+def test_load_metadata_location(populated_role_root):
+    os.rename(
+        os.path.join(populated_role_root, "meta", "main.yml"),
+        os.path.join(populated_role_root, "meta", "main.yaml"),
+    )
+
+    data = LegacyRoleLoader(populated_role_root, "my-namespace").load()
+
+    assert data.namespace == "my-namespace"
+
+    os.rename(
+        os.path.join(populated_role_root, "meta", "main.yaml"),
+        os.path.join(populated_role_root, "meta.yml"),
+    )
+
+    data = LegacyRoleLoader(populated_role_root, "my-namespace").load()
+
+    assert data.name == "my_role"
+
+    os.rename(
+        os.path.join(populated_role_root, "meta.yml"),
+        os.path.join(populated_role_root, "meta.yaml"),
+    )
+
+    data = LegacyRoleLoader(populated_role_root, "my-namespace").load()
+
+    assert data.readme_file == "README.md"
+
+
+def test_load_metadata_invalid_metadata(populated_role_root):
+    with open(os.path.join(populated_role_root, "meta", "main.yml"), "w") as fh:
+        fh.write("")
+
+    with pytest.raises(ValueError, match="must be in the form of a yaml dictionary"):
+        LegacyRoleLoader(populated_role_root, "my-namespace").load()
+
+    with open(os.path.join(populated_role_root, "meta", "main.yml"), "w") as fh:
+        fh.write("hello: person\nanother: field\n")
+
+    with pytest.raises(ValueError, match="galaxy_info field not found"):
+        LegacyRoleLoader(populated_role_root, "my-namespace").load()
+
+    with open(os.path.join(populated_role_root, "meta", "main.yml"), "w") as fh:
+        fh.write("galaxy_info: nope\n")
+
+    with pytest.raises(ValueError, match="galaxy_info field must contain a dictionary"):
+        LegacyRoleLoader(populated_role_root, "my-namespace").load()
+
+    with open(os.path.join(populated_role_root, "meta", "main.yml"), "w") as fh:
+        fh.write("galaxy_info: \n  role_nam: my_role\n")
+
+    with pytest.raises(ValueError, match="unknown field in galaxy_info"):
+        LegacyRoleLoader(populated_role_root, "my-namespace").load()
+
+
+def test_load_name_no_role_name(populated_role_root):
+    with open(os.path.join(populated_role_root, "meta", "main.yml"), "w") as fh:
+        fh.write("galaxy_info: \n  author: me\n")
+
+    data = LegacyRoleLoader(populated_role_root, "my-namespace").load()
+
+    assert data.name == "my_role"
+
+
+@pytest.mark.parametrize(
+    "invalid_name",
+    [
+        "_this",
+        "walker-turzai",
+        "foo_bar-baz",
+        "3w6",
+        "$@#",
+        "this.role",
+        "docker!",
+        "big space",
+        "whyUpper",
+    ],
+)
+def test_load_name_regex(populated_role_root, invalid_name):
```

Comment:
is there a reason you want to test this here, and then run the same tests in `test_schema_legacy`? ... i know this is testing the error is present when calling `LegacyRoleLoader()` vs the other test `LegacyGalaxyInfo()` but my first thought is to keep single DRY category of check closest to the source (in the schema tests)

Reply:
ah, i see there is a regex match against the role name in both the loader and the schema... do we need that check in 2 places?

Reply:
The check in in 2 places because of how we decided to determine the role name. The regex match in the schema will validate the `role_name` metadata field if it exists. Then, in the loader, that becomes the role name. However, if there is no `role_name` metadata field, then the regex match in the schema doesn't run, and the role name inherits from the directory name. This then needs to be regex validated in the loader

Reply:
I can indent the regex match in `LegacyRoleLoader` underneath the `else` block, so that way, it runs only when necessary (`role_name` field not present) and the regex match never runs twice

Reply:
do you see any issues with moving that check out of `LegacyRoleLoader` and into the `LegacyImportResult`?

That way even though we have 2 checks, each is done in the corresponding schema section where the data orginates: `LegacyImportResult` and `LegacyGalaxyInfo`

Reply:
That makes sense. Then, for the namespace regex match, that _could_  be moved to `LegacyImportResult` as well ... but I think by this metric, it still belongs in `LegacyRoleLoader` because namespace is passed directly to `LegacyRoleLoader`